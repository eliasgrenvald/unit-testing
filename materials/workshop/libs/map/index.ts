export * from './base';
export * from './models';
export * from './services';
