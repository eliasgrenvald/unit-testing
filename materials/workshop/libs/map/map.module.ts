import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { Angulartics2Module } from 'angulartics2';
import { MapService } from './services';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import * as fromMapReducers from '@maersk-digital/root-features/map/store/reducers';
import * as fromMapEffects from '@maersk-digital/root-features/map/store/effects';

@NgModule({
	imports:
		[
			CommonModule,
			Angulartics2Module,
			StoreModule.forFeature('map', fromMapReducers.mapReducers),
			EffectsModule.forFeature([
				...fromMapEffects.MAP_EFFECTS
			])
		],
	providers:
		[
			MapService
		]
})
export class MapModule {}
