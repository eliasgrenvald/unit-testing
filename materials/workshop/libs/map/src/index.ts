export * from './lib/map.module';
