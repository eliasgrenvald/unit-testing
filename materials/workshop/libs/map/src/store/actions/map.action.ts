import { Action } from '@ngrx/store';
import { Feature, Point } from 'geojson';

export const SELECT_MAP_FEATURE = '[Testing Workshop] Select A Map Feature';
export const LOAD_MAP_FEATURES = '[Testing Workshop] Load Map Features';
export const LOAD_MAP_FEATURES_SUCCESS = '[Testing Workshop] Load Map Features Succeed';


export class SelectMapFeature implements Action {
	public readonly type = SELECT_MAP_FEATURE;
	public constructor(public readonly payload: Feature<Point>) {}
}

export class LoadMapFeatures implements Action {
	public readonly type = LOAD_MAP_FEATURES;
	public constructor() {}
}

export class LoadMapFeaturesSuccess implements Action {
	public readonly type = LOAD_MAP_FEATURES_SUCCESS;
	public constructor(public readonly payload: ReadonlyArray<Feature<Point>>) {}
}

export type MapAction =
	| SelectMapFeature
	| LoadMapFeatures
	| LoadMapFeaturesSuccess;
