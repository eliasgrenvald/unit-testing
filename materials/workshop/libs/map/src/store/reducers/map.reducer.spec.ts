import { Feature, Point } from 'geojson';
import { SelectMapFeature } from '../actions';
import { reducer, initialState } from './map.reducer';

describe('SelectMapFeature action', () => {
	test('should set selectedFeature', () => {
		const point = <Feature<Point>>{
			type: 'Feature',
			geometry:
				{
					type: 'Point',
					coordinates:
						[
							10.263290405273438,
							56.17155205794858
						]
				}
		};

		const action = new SelectMapFeature(point);
		const result = reducer(initialState, action);

		expect(result).toMatchSnapshot();
	});
});