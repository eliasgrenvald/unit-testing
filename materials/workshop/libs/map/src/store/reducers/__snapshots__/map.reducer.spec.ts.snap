// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectMapFeature action should set selectedFeature 1`] = `
Object {
  "features": Array [],
  "selectedFeature": Object {
    "geometry": Object {
      "coordinates": Array [
        10.263290405273438,
        56.17155205794858,
      ],
      "type": "Point",
    },
    "type": "Feature",
  },
}
`;
